### ISSUE TITLE
Occasional ROT.Map.Rogue crashes on rotLove.lua:3622

### ISSUE THREAD

**Issue by globalcitizen** [paulofmandown/rotLove#10]
The setup.
```
cl=ROT.Map.Rogue:new(resolutionTilesX, resolutionTilesY)
cl:create(tile_callback)
```

The **rotLove.lua** line in question, inside of **function ROT.Map.Rogue:_getWallPosition(aRoom, aDirection)**:
```
        self.map[door][ry]=0
```

The crash.
```
-- Unable to connect room.
Error: rotLove.lua:3622: table index is NaN
stack traceback:
       	rotLove.lua:3622: in function '_getWallPosition'
       	rotLove.lua:3694: in function '_createCorridors'
       	rotLove.lua:3405: in function 'create'
       	main.lua:106: in function 'load'
       	[string "boot.lua"]:439: in function <[string "boot.lua"]:435>
       	[C]: in function 'xpcall'
```

**Comment by globalcitizen** [comment]
And again. This time without the **-- Unable to connect room** part, which is occasional and apparently sorts itself on retry automagically.

```
Error: rotLove.lua:3622: table index is NaN
stack traceback:
       	rotLove.lua:3622: in function '_getWallPosition'
       	rotLove.lua:3694: in function '_createCorridors'
       	rotLove.lua:3405: in function 'create'
       	main.lua:108: in function 'load'
       	[string "boot.lua"]:439: in function <[string "boot.lua"]:435>
       	[C]: in function 'xpcall'
```

FYI the tilemap size I am passing in is 90 x 56 tiles, which is 1440 x 900 pixels fullscreen divided by a 16 x 16 pixel tilesize.

**Comment by globalcitizen** [comment]
On the subject of doors I would add that 3-way corridor doors do seem to be generated now and then. Not sure about 4-way, haven't seen one of those yet.

**Comment by paulofmandown** [comment]
This should be fixed in 68365b5. With the dimensions you provided, there was the potential for 1 tile tall rooms. _getWallPosition didn't account for this and would then call _getRandomInt with a max that was lower than min.
